### PR TITLE
fix(table): check length of tableModel.data before select/deselect all

### PR DESCRIPTION
**Pull request by Donisius**
When the table model contains no data and onSelectAll is called, it gives the following console error:

```
ERROR TypeError: Cannot read property 'length' of undefined
    at TableModel.push../src/table/table-model.class.ts.TableModel.selectAll (VM37 main.5f83aacd3a20d62a5fc8.bundle.js:21725)
    at Table.push../src/table/table.component.ts.Table.onSelectAll (VM37 main.5f83aacd3a20d62a5fc8.bundle.js:22320)
    at Object.eval [as handleEvent] (VM104 Table.ngfactory.js:107)
    at handleEvent (VM36 vendors~main.5f83aacd3a20d62a5fc8.bundle.js:47008)
    at callWithDebugContext (VM36 vendors~main.5f83aacd3a20d62a5fc8.bundle.js:48101)
    at Object.debugHandleEvent [as handleEvent] (VM36 vendors~main.5f83aacd3a20d62a5fc8.bundle.js:47804)
    at dispatchEvent (VM36 vendors~main.5f83aacd3a20d62a5fc8.bundle.js:44467)
    at VM36 vendors~main.5f83aacd3a20d62a5fc8.bundle.js:45947
    at SafeSubscriber.schedulerFn [as _next] (VM36 vendors~main.5f83aacd3a20d62a5fc8.bundle.js:40317)
    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub (VM36 vendors~main.5f83aacd3a20d62a5fc8.bundle.js:137132)

```

it now behaves as expected.

### CODE REVIEW

File: src/table/table.component.ts [hunk 1]
```diff
@@ -628,13 +628,17 @@ export class Table implements AfterViewInit, OnDestroy {
 	}
 
 	onSelectAll() {
-		this.model.selectAll(true);
-		this.selectAll.emit(this.model);
+		if (this.model.data.length > 1) {
+			this.model.selectAll(true);
```
Comment (cal-smith) [inline src/table/table.component.ts]: Maybe it makes sense for the model to handle this? If there's no data `selectAll` would still select all the non-existent data right?

Comment (Donisius) [reply src/table/table.component.ts]: Yeah, that makes a lot more sense

File: src/table/table-model.class.ts [hunk 2]
```diff
@@ -522,8 +522,10 @@ export class TableModel implements PaginationModel {
 	 * @param value state to set all rows to. Defaults to `true`
 	 */
 	selectAll(value = true) {
-		for (let i = 0; i < this.rowsSelected.length; i++) {
-			this.selectRow(i, value);
+		if (this.data.length > 1) {
```
Comment (theiliad) [inline src/table/table-model.class.ts]: Shouldn't it be `if (this.data.length > 0) {`? When the `length` is 1 you should still be able o select all

Comment (Donisius) [reply src/table/table-model.class.ts]: If there is no data in the model, it sets a single item so length is 1 when it's empty:
```
set data(newData: TableItem[][]) {
		if (!newData || (Array.isArray(newData) && newData.length === 0) ) {
			newData = [[]];
		}
```

**Pull request merged**